c/tskit/tables.c

```diff
@@ -643,6 +643,40 @@ tsk_individual_table_truncate(tsk_individual_table_t *self, tsk_size_t num_rows)
     return ret;
 }
 
+int
+tsk_individual_table_extend(tsk_individual_table_t *self,
+    const tsk_individual_table_t *other, tsk_size_t num_rows,
+    const tsk_id_t *row_indexes, tsk_flags_t TSK_UNUSED(options))
+{
+    int ret = 0;
+    tsk_size_t j;
+    tsk_individual_t individual;
+
+    /* We know how much to expand the non-ragged columns, so do it ahead of time */
+    if (self->max_rows < self->num_rows + num_rows) {
+        ret = tsk_individual_table_expand_main_columns(self, num_rows);
```

Comment:
Ah, sorry, should have spotted this the first time around. You can just call ``ret = tsk_individual_table_expand_main_columns(self, num_rows);`` without the first check. It does the check to see if there's space for the specified number of additional rows, and expands if necessary.

Reply:
Of course! Fixed.